src/proof/uf_proof.cpp

```diff
@@ -649,11 +649,13 @@ void LFSCUFProof::printOwnedTermAsType(Expr term,
     os << "(apply _ _ ";
   }
   os << func << " ";
+  Assert(func.getType().isFunction());
+  FunctionType funcType = static_cast<FunctionType>(func.getType());
   for (unsigned i = 0; i < term.getNumChildren(); ++i) {
 
     bool convertToBool = (term[i].getType().isBoolean() && !d_proofEngine->printsAsBool(term[i]));
     if (convertToBool) os << "(f_to_b ";
-    d_proofEngine->printBoundTerm(term[i], os, map);
+    d_proofEngine->printBoundTerm(term[i], os, map, TypeNode::fromType(funcType.getArgTypes()[i]));
```

Comment:
How about computing `funcType.getArgTypes()` before the loop?

Reply:
This shouldn't be necessary. Its a const method and the compiler should be able to lift it automatically.

Reply:
I seem to recall discussing something similar with @aniemetz and would be interested in her opinion on this! I am not sure what assumptions do we adopt about the compiler.

Reply:
We don't want (never) to rely on the compiler here (we don't have control over the compiler, and which compiler is used). And even if it is static, you could have side effects that make it impossible to lift. It's best practice to move such function calls to outside of the loop.

Reply:
I agree with @aniemetz. It seems unlikely that the compiler could do this here and even if it could, it's best not to rely on it.